### PR TITLE
Update consistent slicing RFC with new proposal

### PR DESCRIPTION
**Pull request by nelson-wu**
PR at https://github.com/dotnet/fsharp/pull/7541

### CODE REVIEW

File: FSharp.Core-4.4.3.0/FS-1036-consistent-slicing.md [hunk 1]
```diff
@@ -35,10 +39,55 @@ The flaw in `l.[..j]` generates **unexpected runtime errors**. E.g.
...
+`L[x:y]`  | a <= x <= b+1 | x > b+1
+-----------------|------------------|------
+a <= y <= b+1 | if x > y [] else `{ L[x] .. L[y-1] }` | []
+y > b+1 |  `{ L[x] .. L[b] }` | []
+
+(Note: x < a and y < a cases aren't shown here because Python negative indices mean something else)
+
+---
+
+As in the example above the slicing behavior in F# is a lot more confusing than C# or Python. Sometimes when indexes are out of bounds you get [], sometimes you get Error, and sometimes you get either.
```
[the remark this answers is not in the document]
Comment (nelson-wu) [reply FSharp.Core-4.4.3.0/FS-1036-consistent-slicing.md]: That's a good point. I've updated the RFC with an example on how this new behavior can be used to simplify machine learning code.

File: FSharp.Core-4.4.3.0/FS-1036-consistent-slicing.md [hunk 2]
```diff
@@ -3,23 +3,27 @@
...
-* [ ] Implementation: [Complete](https://github.com/Microsoft/visualfsharp/pull/3475)
+* [ ] Implementation: [Completed PR #1](https://github.com/Microsoft/visualfsharp/pull/3475) | [In Progress PR #2](https://github.com/dotnet/fsharp/pull/7541)
 
 
 # Summary
 [summary]: #summary
 Currently, if `x` is a list, `x.[..j]` gives an out of bounds exception when j<0, while `x.[0..j]` gives `[]`.
 This proposes changing `x.[..j]` to give the same output as `x.[0..j]`, and `x.[i..]` to give the same output as `x.[i..(x.Length-1)]`
 
+In addition, there is some inconsistency in the out-of-bounds behavior for slicing these collections as well. There is an opportunity here to make the out-of-bounds slicing behavior more consistent and predictable for users.
```
[the remark this answers is not in the document]
Comment (charlesroddie) [reply FSharp.Core-4.4.3.0/FS-1036-consistent-slicing.md]: Not resolved.

**Comment by dsyme** [comment]
@cartermp IIUC we can at least pull this PR and then assess the overall RFC?  That is, if we do this change, the one described by this PR is the one we want??

**Comment by charlesroddie** [comment]
@dsyme The original RFC was approved so if we don't do the changes in this PR then it would be reverting to the original RFC. If it's going to be pulled now for review, could it at least be split into two parts, original and supplement, as these do different things? The changes are separate ("consistent slicing" bugfix in original RFC, "tolerant slicing" new proposal), can be evaluated separately, and the arguments in the two parts are not yet compatible, so it's a mess to put them together.

**Comment by cartermp** [comment]
@dsyme I agree, though it's worth noting that your statement about liking _both_ designs is technically covered here since one is a superset of another. So you've technically approved this design. Is that your intention?

**Comment by dsyme** [comment]
@nelson-wu This should be a separate RFC please.  [RFC FS-1036](https://github.com/fsharp/fslang-design/blob/master/FSharp.Core-4.4.3.0/FS-1036-consistent-slicing.md) is closed and done and delivered and we shouldn't update it further. Thank you!

**Pull request closed without merging**